pandas/core/groupby/groupby.py

```diff
@@ -1887,12 +1888,10 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                                      fill_method=fill_method,
                                                      limit=limit, freq=freq,
                                                      axis=axis))
-
         with _group_selection_context(self) as new:
```

Comment:
Unlike in the series groupby, the context manager here successfully resets the cache, although it provides no solution for preserving the state.